index.js

```diff
@@ -64,12 +64,16 @@ function get(url, timeout, port, protocol, detailed) {
   protocol = protocol || 'https:';
 
   var options = getOptions(url, port, protocol);
+  
+  if (timeout) {
+    options['timeout'] = timeout;
+  }
 
   return new Promise(function(resolve, reject) {
     var req = handleRequest(options, detailed, resolve, reject);
 
     if (timeout) {
-      req.setTimeout(timeout, function() {
+      req.on('timeout', () => {
```

Comment:
I would not use that as this works differently (just an event handler, you have to clean up after it).

https://nodejs.org/api/http.html#http_event_timeout

> Emitted when the underlying socket times out from inactivity. This only notifies that the socket has been idle. The request must be aborted manually.